core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1856,113 +1627,38 @@ class WorkspaceService(
-  private def withWorkspaceContext[T](workspaceName: WorkspaceName,
-                                      dataAccess: DataAccess,
-                                      attributeSpecs: Option[WorkspaceAttributeSpecs] = None
-  )(op: (Workspace) => ReadWriteAction[T]) =
-    dataAccess.workspaceQuery.findByName(workspaceName, attributeSpecs) flatMap {
-      case None            => throw NoSuchWorkspaceException(workspaceName)
-      case Some(workspace) => op(workspace)
-    }
+  def checkSamActionWithLock(workspaceName: WorkspaceName, samAction: SamResourceAction): Future[Boolean] =
+    getV2WorkspaceContextAndPermissions(workspaceName, samAction, None)
+      .map(_ => true)
+      .recover { case _ => false }
 
   // Finds workspace by workspaceName
   // moved out of WorkspaceSupport because the only usage was in this file,
   // and it has raw datasource/dataAccess usage, which is being refactored out of WorkspaceSupport
   private def withV2WorkspaceContext[T](workspaceName: WorkspaceName,
                                         dataAccess: DataAccess,
                                         attributeSpecs: Option[WorkspaceAttributeSpecs] = None
-  )(op: (Workspace) => ReadWriteAction[T]) =
+  )(op: Workspace => ReadWriteAction[T]) =
     dataAccess.workspaceQuery.findV2WorkspaceByName(workspaceName, attributeSpecs) flatMap {
       case None            => throw NoSuchWorkspaceException(workspaceName)
       case Some(workspace) => op(workspace)
     }
 
-  def getBucketUsage(workspaceName: WorkspaceName): Future[BucketUsageResponse] =
-    // don't do the sam REST call inside the db transaction.
-    getV2WorkspaceContext(workspaceName) flatMap { workspaceContext =>
-      requireAccessIgnoreLockF(workspaceContext, SamWorkspaceActions.read) {
-        // if we get here, we passed all the hoops, otherwise an exception would have been thrown
-        gcsDAO.getBucketUsage(workspaceContext.googleProjectId, workspaceContext.bucketName).recoverWith {
-          // Throw with the status code of the google exception (for example 403 for invalid billing, 404 for inactive project)
-          // instead of a 500 to avoid Sentry notifications.
-          case t: GoogleJsonResponseException =>
-            val code = getStatusCodeHandlingUnknown(t.getStatusCode)
-            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(code, t.getDetails.toString)))
-        }
-      }
-    }
+  def getBucketOptions(workspaceName: WorkspaceName): Future[WorkspaceBucketOptions] = for {
+    workspaceContext <- getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read)
+    options <- gcsDAO.getBucketDetails(workspaceContext.bucketName, workspaceContext.googleProjectId)
+  } yield options
+
+  def getBucketUsage(workspaceName: WorkspaceName): Future[BucketUsageResponse] = (for {
+    workspaceContext <- getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read)
```

Comment:
This used to ignore the lock, but it doesn't anymore because `read` isn't blocked by a lock anyways?

Reply:
Correct.